### PR TITLE
bug: fixes #25 add cleanup in scoremeter 

### LINKED ISSUE #25: Can't perform a state update on an unmounted Scoremeter component
image

### PR DESCRIPTION
**Pull request by shanpriyan**
Added cleanup in useEffect

### CODE REVIEW

File: src/components/Scoremeter/index.tsx [hunk 1]
```diff
@@ -49,9 +49,9 @@ const ScoreMeter = (props: ScoremeterProps) => {
     }, []);
 
     React.useEffect(() => {
-        let timerId;
+        let timerId: number;
         if (score !== sanitizedReading) {
-            timerId = setTimeout(() => {
+            timerId = window.setTimeout(() => {
```
Comment (harish-sethuraman) [inline src/components/Scoremeter/index.tsx:54]: But setTimeout id browser API right? we can use it directly too?  what is the catch here?

File: src/components/Scoremeter/index.tsx [hunk 2]
```diff
@@ -49,20 +49,28 @@ const ScoreMeter = (props: ScoremeterProps) => {
     }, []);
 
     React.useEffect(() => {
+        let timerId: number;
         if (score !== sanitizedReading) {
-            setTimeout(() => {
+            timerId = window.setTimeout(() => {
```
Comment (harish-sethuraman) [inline src/components/Scoremeter/index.tsx:54]:
```suggestion
            const timerId = window.setTimeout(() => {
```

File: src/components/Scoremeter/index.tsx [hunk 3]
```diff
@@ -49,20 +49,28 @@ const ScoreMeter = (props: ScoremeterProps) => {
     }, []);
 
     React.useEffect(() => {
+        let timerId: number;
```
Comment (harish-sethuraman) [inline src/components/Scoremeter/index.tsx:52]:
```suggestion
```

File: src/components/Scoremeter/index.tsx [hunk 4]
```diff
@@ -49,20 +49,28 @@ const ScoreMeter = (props: ScoremeterProps) => {
...
                 setScoreUpdate(true);
             }, 500);
         }
+        return () => {
+            clearTimeout(timerId);
+        };
     }, [animate]);
 
     React.useEffect(() => {
+        let timerId: number;
```
Comment (harish-sethuraman) [inline src/components/Scoremeter/index.tsx:64]:
```suggestion
```

File: src/components/Scoremeter/index.tsx [hunk 5]
```diff
@@ -49,20 +49,28 @@ const ScoreMeter = (props: ScoremeterProps) => {
...
         if (score !== sanitizedReading) {
-            setTimeout(() => {
+            timerId = window.setTimeout(() => {
                 setScoreUpdate(true);
             }, 500);
         }
+        return () => {
+            clearTimeout(timerId);
```
Comment (harish-sethuraman) [inline src/components/Scoremeter/index.tsx:59]: If `score !== sanitizedReading` we might get the timerId? but if not are we doing a `clearTimeout(undefined)`

File: src/components/Scoremeter/index.tsx (hunk 1 above)
Comment (shanpriyan) [reply src/components/Scoremeter/index.tsx:54]: After making the `timerId` type number, typescript expects that `setTimeout` returns a `Timeout` object as in Node. In order to let typescript know that we are using `setTimeout` from browser API,  added **window** before setTimeout.
https://stackoverflow.com/a/55550147/17948948

The other solution would be adding ''type" with an empty array in `tsconfig.json`
```js
{
  "compilerOptions": {
    "types": []
  }
}
```
https://stackoverflow.com/a/70856713/17948948

File: src/components/Scoremeter/index.tsx (hunk 5 above)
Comment (shanpriyan) [reply src/components/Scoremeter/index.tsx:59]: Yes

**Review by harish-sethuraman — APPROVED** [review]
LGTM

**Review by tripurari001 — APPROVED** [review]

**Pull request merged**